util/logger/src/lib.rs

```diff
@@ -57,6 +58,15 @@ macro_rules! error {
     }
 }
 
+/// Enable metrics collection feature by setting `ckb-metrics=trace` in logger filter.
+#[macro_export(local_inner_macros)]
+macro_rules! metric {
+    ($( $args:tt )*) => {
+        let output = $crate::json!($( $args )*);
+        $crate::internal::trace!(target: "ckb-metrics", "{}", output);
```

Comment:
This PR use "measurement": "error" everywhere, if the output does not include `CARGO_PKG_NAME`, it will be hard to distinguish these metrics.

I suggest attach `"package": "CARGO_PKG_NAME"` to the output json. Or add a prefix `${CARGO_PKG_NAME}_error` to "measurement" fields.

Reply:
Also, IMO the "measurement" is not a good name, we can just use "name" or "topic" as the key.

Reply:
> This PR use "measurement": "error" everywhere, if the output does not include `CARGO_PKG_NAME`, it will be hard to distinguish these metrics.
> 
> I suggest attach `"package": "CARGO_PKG_NAME"` to the output json. Or add a prefix `${CARGO_PKG_NAME}_error` to "measurement" fields.

Adding `"package": "CARGS_PKG_NAME"` is a good filter to identify the source of metrics!  We can reuse "logger target" as the package names.

Reply:
> Also, IMO the "measurement" is not a good name, we can just use "name" or "topic" as the key.

Here use the keywords as same as https://docs.influxdata.com/influxdb/v1.7/write_protocols/line_protocol_tutorial/
Could you please explain why "name"/"topic" is better than "measurement"?

Reply:
Reusing the log target as measure name is a good idea. We can use `measure.{measure_name}` as the logger target. The log body only need to include measure value and tags.

Reply:
This way we can enable/disable each measure via logger filter

Reply:
"name"/"topic" is shorter, easier to type, no chance typo.

Do we have any reason to use the same key to a specific backend DB?

Reply:
> Reusing the log target as measure name is a good idea. We can use `metric.{metric_name}` as the logger target. The log body only need to include measure value and tags.

But how do we enable all metrics? Does RUST_LOG support this `key.*=trace`?

Reply:
I don't think we have the requirements to enable/disable specific measures.
All or never.
I think that adding `tags: { "package": LOGGER_TARGET }` is just for distinguishing and filtering.

Reply:
Update in last commit 630ad552e10867a7275c131f730cb4eaa37c4022
    * Remove useless field `{"count": 1}`
    * Add new tag `{"target": crate::env!("CARGO_PKG_NAME")}` by default
    * Rename "measurement" to "topic"
    * Rename "source" to "input"
    * Rename "warn" to "warning"

Please take a look. Thanks.